### DISCUSSION TITLE
What commands would you like to see for Codeline?

### DISCUSSION THREAD

**Discussion by rorybyrne** [synek/codeline#18] [Q&A]
Right now, the two I've had in mind are `commit` for committing to git, and `codegen` for generating code via some API.

**Comment by yasyf** [comment]
running / generating a test for a given function

**Reply by rorybyrne** [reply]
Love the idea of running a test (or even the function itself if you specify input args).

**Comment by yasyf** [comment]
(auto?) generating docs / scaffolding for a given function

**Comment by yasyf** [comment]
populating type annotations for a function signature

**Comment by shelper** [comment]
just curious how this codegen command is implemented?
is there a tool for this?
I googled and didnt find any

**Reply by rorybyrne** [reply]
Hey, the `codegen` command doesn't exist yet. The GIF is just a hard-coded example.

There are some startups doing that kind of thing though, e.g. https://debuild.co.

Hopefully it will be easy to integrate with Codeline via an API call.

**Reply by rorybyrne** [reply]
As far as I know, Debuild uses OpenAI's GTP-3 API under the hood.

**Comment by rorybyrne** [comment]
An idea I had today:

Point `ngrok` at a specific function:

```
# <| ngrok -d "my-subdomain"
def func(width, height):
    return {
        "result": width * height
    }
```

Then run `curl https://my-subdomain.ngrok.io?width=100&height=50`, and the response body is a JSON string `{ "result": 5000 }`

**Comment by rorybyrne** [comment]
Another use-case:

Keeping a file in-sync with another version

```
"""Module-level docstring

<| file-sync https://github.com/myorg/myrepo/tree/master/somefile.py
"""

CONSTANT_ONE = "blah"
CONSTANT_TWO = "blah blah"
```

Run this as a pre-commit hook to keep some stuff updated?
